libsignal-service/src/account_manager.rs

```diff
@@ -1,14 +1,32 @@
-use crate::{TrustStore, USER_AGENT};
+use crate::{
+    push_service::SmsVerificationCodeResponse, TrustStore, USER_AGENT,
+};
+
+use async_trait::async_trait;
 use failure::Error;
 
-pub struct AccountManager;
+#[async_trait(?Send)]
+pub trait AccountManager {
+    fn builder() -> AccountManagerBuilder { AccountManagerBuilder::default() }
```

Comment:
90% of the time you don't want to specify how something is constructed as part of a trait. That's because now you're contractually obliged to only ever use the information inside `AccountManagerBuilder`... For example, what if I need to pass in a the mailbox to another actor, or a reference to an event loop (tokio or Qt) that I can spawn callbacks on?

Instead of the `builder()` and `from_builder()` methods I'd leave it up to the concrete type to determine how it is constructed, and use trait methods to only define required behaviour (e.g. `request_sms_verification_code()`).

Reply:
I've gotten rid of the forced constructors, but that also means the `AccountManagerBuilder` was kinda out of place (since the AccountManager is now built from a `PushService`).